### ISSUE TITLE
[GPU] ProgramBuilder fails on Select ops with 5D tensors from PyTorch traced .masked_fill() / .tril()

### ISSUE THREAD

**Issue by t8** [openvinotoolkit/openvino#35271]
## OpenVINO Version

2026.0.0-20965-c6d6a13a886-releases/2026/0

## Operating System

Windows 11 (Intel Core Ultra X7 358H / Panther Lake iGPU)

## Device used for inference

GPU

## Framework

PyTorch (torch.jit.trace → ov.convert_model)

## Model used

Minimal reproducer below. Also observed in real models: Qwen3.5-27B GatedDeltaNet layers, which use `.masked_fill()` on 5D `[batch, heads, chunks, chunk_size, chunk_size]` attention tensors.

## Issue description

GPU compilation fails when the OV graph contains `Select` ops on **5D tensors** produced by PyTorch-traced `.masked_fill()` or `.tril()`. The same Select ops on 2D, 3D, and 4D tensors compile successfully.

The error originates from `select_kernel_base.cpp`'s `MASK` JIT macro being expanded inside the `select()` call in `select_gpu_ref.cl`. For 5D inputs, the MASK expansion includes `INPUT0_GET_INDEX_SAFE(...)` with the wrong number of arguments:

```
const OUTPUT_TYPE res = TO_OUTPUT_TYPE(select(INPUT_2, INPUT_1, MASK));
                                                                 ^
note: expanded from macro 'MASK'
error: use of undeclared identifier 'INPUT0_GET_INDEX_SAFE'
```

### Rank sweep results

| Rank | Shape | .masked_fill() | .tril() |
|------|-------|----------------|---------|
| 2D | (16, 16) | OK | — |
| 3D | (1, 16, 16) | OK | OK |
| 4D | (1, 4, 16, 16) | OK | OK |
| **5D** | **(1, 4, 2, 16, 16)** | **FAILS** | **FAILS** |
| 6D | (1, 2, 2, 2, 16, 16) | FAILS (layout) | — |

### Workaround

Replace `.masked_fill(mask, value)` or `.tril()` with multiplication by a pre-computed constant lower triangular mask. This eliminates `Select` ops from the OV graph entirely:

```python
# Before (fails on GPU for 5D tensors):
x = x.masked_fill(upper_tri_bool_mask, 0.0)
# or
x = x.tril()

# After (works on GPU):
lower_tri = torch.tril(torch.ones(S, S, dtype=torch.float32))
x = x * lower_tri  # broadcasts over batch/head/chunk dims
```

### Root cause

`select_gpu_ref.cl` defines `INPUT_0`, `INPUT_1`, `INPUT_2` macros using `OUTPUT_DIMS` to determine how many index arguments to pass to each input's `GET_INDEX_SAFE` macro. But `GET_INDEX_SAFE` is generated per-tensor in `jitter.cpp` based on that tensor's actual layout — so a 4D input's `GET_INDEX_SAFE` accepts 4 arguments (`b, f, y, x`) while the 5D output's `#if OUTPUT_DIMS == 5` block passes 5 (`b, f, z, y, x`).

This breaks whenever an input has fewer dimensions than the output — the typical case being a broadcast condition mask.

### Suggested fix

Move `INPUT_N` macro generation from the `.cl` template into `select_kernel_base.cpp`, where each input's actual dimensionality (`Dimentions()`) can be used to select the correct index arguments. Will submit a PR with this fix.

## Step-by-step reproduction

```python
import torch, torch.nn as nn, openvino as ov

class MaskedFillModel(nn.Module):
    def __init__(self, S):
        super().__init__()
        mask = torch.triu(torch.ones(S, S, dtype=torch.bool), diagonal=1)
        self.register_buffer("mask", mask)

    def forward(self, x):
        return x.masked_fill(self.mask, 0.0)

model = MaskedFillModel(16).eval()
x = torch.randn(1, 4, 2, 16, 16)  # 5D tensor
traced = torch.jit.trace(model, (x,))
ov_model = ov.convert_model(traced, example_input=(x,))

# Fails with MASK macro conflict:
compiled = ov.Core().compile_model(ov_model, "GPU")
```

## Relevant log output

```
Exception from src\inference\src\cpp\core.cpp:113:
Exception from src\inference\src\dev\plugin.cpp:53:
Check 'false' failed at src\plugins\intel_gpu\src\plugin\program_builder.cpp:163:
[GPU] ProgramBuilder build failed!
Program build failed(0_part_0):
1:4934:66: error: too many arguments provided to function-like macro invocation
 const OUTPUT_TYPE res = TO_OUTPUT_TYPE(select(INPUT_2, INPUT_1, MASK));
                                                                 ^
1:4890:36: note: expanded from macro 'MASK'
....
1:4934:66: error: use of undeclared identifier 'INPUT0_GET_INDEX_SAFE'
 const OUTPUT_TYPE res = TO_OUTPUT_TYPE(select(INPUT_2, INPUT_1, MASK));
                                                                 ^
1:4890:36: note: expanded from macro 'MASK'
```

## Related issues

- #33618 — GPU ProgramBuilder failure on Clip with UINT8 (same class: OpenCL kernel macro expansion)
- #30977 — ProgramBuilder failure from driver version (different root cause)

## Checklist

- [x] I have searched existing issues and this has not been reported
- [x] This is a bug, not a question
- [x] Reproducer code is provided
